libass/ass_fontselect.c

```diff
@@ -166,6 +170,111 @@ static ASS_FontProviderFuncs ft_funcs = {
     .destroy_font      = destroy_font_ft,
 };
 
+#if defined(_WIN32) && !defined(__CYGWIN__)
+/*
+ * Output parameters are only valid if the call was sucessful, ignore them otherwise.
+ * \param path          input; path of the dir to open, may be UTF-8 or the codepage used by fopen
+ * \param file_context  output
+ * \param find_handle   output; needs to be closed
+ * \param path_utf8     output; utf-8 encoded version of path; needs to be freed
+ * \return true if succesful, false otherwise
+ */
+static wchar_t *win_open_dir(const char *path, WIN32_FIND_DATAW *file_context, HANDLE *find_handle, size_t *out_length)
+{
```

Comment:
The documentaion no longer matches the signature and implementation.

Also, instead of using a return value **and additionally** an output-argument, it's better to use a single `ASS_StringView` from `ass_utils.h`, as MrSmile suggested.
The latter applies to all function changed this way.

Reply:
Oups, forgot this is a `wchar_t*`, so `ASS_StringView doesn't work (except for the conversion **to** UTF-8). And declaring a new Windows-exclusive type isn't too pretty either; nvm then.